FlippersDelayed/src/main/java/io/doist/recyclerviewext/flippers/DelayedEmptyRecyclerFlipper.kt

```diff
@@ -0,0 +1,66 @@
+package io.doist.recyclerviewext.flippers
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.IdRes
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.recyclerview.widget.RecyclerView
+
+/**
+ * Similar to [EmptyRecyclerFlipper], but delays the empty state appearance.
+ *
+ * An example of usage is a View that contains a list of local data that can take
+ * a few milliseconds to load and has an entrance animation. With the delay being the same as
+ * the animation, the user won't ever see the empty state view during the animation.
+ */
+open class DelayedEmptyRecyclerFlipper(
+    recyclerView: RecyclerView,
+    emptyView: View,
+    lifecycleOwner: LifecycleOwner,
+    private val delayedFlipper: DelayedFlipper = DelayedFlipper(lifecycleOwner)
+) : EmptyRecyclerFlipper(recyclerView, emptyView, delayedFlipper) {
+
+    constructor(
+        container: ViewGroup,
+        @IdRes recyclerViewId: Int,
+        @IdRes emptyViewId: Int,
+        lifecycleOwner: LifecycleOwner,
+        delayedFlipper: DelayedFlipper = DelayedFlipper(lifecycleOwner)
+    ) : this(
+        container.findViewById(recyclerViewId),
+        container.findViewById(emptyViewId),
+        lifecycleOwner,
+        delayedFlipper
+    )
+
+    private val lifecycleObserver = object : LifecycleObserver {
+        @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+        private fun onDestroyed() {
+            delayedFlipper.cancelPendingFlip()
+        }
+    }
+
+    init {
+        lifecycleOwner.lifecycle.addObserver(lifecycleObserver)
+    }
```

Comment:
No, I've overlapped the responsibilities of both classes. I've removed the lifecycle observing from `EmptyRecyclerFlipper` and `ProgressEmptuRecyclerFlipper` since it was only listening to what `DelayedFlipper` does.